src/libguac/client.c

```diff
@@ -701,6 +739,116 @@ int guac_client_owner_supports_required(guac_client* client) {
     
 }
 
+/**
+ * A callback function that is invokved by guac_client_owner_notify_join() to
+ * notify the owner of a connection that another user has joined the
+ * connection, returning zero if the message is sent successfully, or non-zero
+ * if an error occurs.
+ *
+ * @param user
+ *     The user to send the notification to, which will be the owner of the
+ *     connection.
+ *
+ * @param data
+ *     The data provided to the callback, which is the user that is joining the
+ *     connection.
+ *
+ * @return
+ *     Zero if the message is sent successfully to the owner, otherwise
+ *     non-zero, cast as a void*.
+ */
+static void* guac_client_owner_notify_join_callback(guac_user* user, void* data) {
+
+    const guac_user* joiner = (const guac_user *) data;
+
+    if (user == NULL)
+        return (void*) ((intptr_t) -1);
+
+    char* owner = "owner";
+    if (user->info.name != NULL)
+        owner = (char *) user->info.name;
+
+    char* joinName = "anonymous";
+    if (joiner->info.name != NULL)
+        joinName = (char *) joiner->info.name;
+
+    guac_user_log(user, GUAC_LOG_DEBUG, "Notifying owner \"%s\" of \"%s\" joining.", owner, joinName);
+    
+    /* Send required parameters to owner. */
+    const char* args[] = { (const char*)joinName, NULL };
```

Comment:
While I agree it makes sense to log something human-readable in the event this is missing ("anonymous", "owner"), I think the value we send via `msg` for `GUAC_MESSAGE_USER_JOINED` and `GUAC_MESSAGES_USER_LEFT` should be strictly the value we receive, or perhaps empty string if `NULL`. If an implementation were to assign its own meaning to the literal `"anonymous"` and send it voluntarily via `name`, then this automatically-assigned value would collide with that.

Reply:
Okay, I've fixed this up, though it may not be as elegant as it could be - let me know if there's a better way to do this...